packages/webapp/src/components/Icons/icons.tsx

```diff
@@ -63,8 +63,29 @@ import { ReactComponent as SoilAmendmentTask } from '../../assets/images/task/So
 import { ReactComponent as MoreHorizontalIcon } from '../../assets/images/more-horizontal.svg';
 import { ReactComponent as PlusCircleIcon } from '../../assets/images/plus-circle.svg';
 import { ReactComponent as TrashIcon } from '../../assets/images/animals/trash_icon_new.svg';
+import { FunctionComponent } from 'react';
 
-const iconMap = {
+const animalIconMap = {
+  CATTLE: CattleIcon,
+  CHICKEN: ChickenIcon,
+  PIGS: PigIcon,
+  BATCH: BatchIcon,
+  BATCH_GREEN: BatchIconGreen, // svgColorFill does not work with this icon
+  CUSTOM_ANIMAL: CustomAnimalIcon,
+  ALPACA: AlpacaIcon,
+  GOAT: GoatIcon,
+  RABBIT: RabbitIcon,
+  SHEEP: SheepIcon,
+};
+
+const AnimalTypeIcons = Object.keys(animalIconMap);
+export type AnimalTypeIconKey = (typeof AnimalTypeIcons)[number];
+// TypeGuard for animal type icons
+export const isAnimalTypeIconKey = (iconKey: string): iconKey is AnimalTypeIconKey => {
+  return AnimalTypeIcons.includes(iconKey as AnimalTypeIconKey);
```

Comment:
So it can't just be removed for me -- causes a type error on iconKey in includes wanting to be of that type. I am mixing two examples:

Has casting:
[Typescript TypeGuard:](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

Uses type `any`:
[StackOverflow example:](https://stackoverflow.com/questions/51528780/typescript-check-typeof-against-custom-type)

So I could remove the cast but then the function prop would want something like `(iconKey: string)` => `(iconKey: any)` and I think typing the string could be valuable for the includes function. So I am not quite sure.

Reply:
Sorry, that was only referring to the version of the code that you reverted!